distributed/tests/test_client.py

```diff
@@ -8029,3 +8030,41 @@ def identity(x):
     outer_future = c.submit(identity, {"x": inner_future, "y": 2})
     result = await outer_future
     assert result == {"x": 1, "y": 2}
+
+
+def bad_wrapper(client, use_worker_client=False):
+    class FuncWrapper:
+        def __init__(self, f, args=None):
+            self.f = f
+            self.args = [] if args is None else args
+
+        def __call__(self, x):
+            return self.f(x, *self.args)
+
+    def make_data():
+        return 1
+
+    def func_to_map(x, x0):
+        if use_worker_client:
```

Comment:
```suggestion
        sleep(0.1)
        if use_worker_client:
```
This seemed to be necessary to reproduce https://github.com/dask/distributed/issues/7746

Reply:
I did reproduce for me with the simpler configuration but I'm happy to add a sleep here